src/main/java/configurationslicing/docker/AbstractDockerSlicerSpec.java

```diff
@@ -0,0 +1,185 @@
+package configurationslicing.docker;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.cloudbees.dockerpublish.DockerBuilder;
+import com.google.common.base.Throwables;
+import com.google.common.collect.Lists;
+import configurationslicing.UnorderedStringSlicer.UnorderedStringSlicerSpec;
+import hudson.matrix.MatrixProject;
+import hudson.model.AbstractProject;
+import hudson.model.Descriptor;
+import hudson.model.Project;
+import hudson.tasks.Builder;
+import hudson.util.DescribableList;
+import jenkins.model.Jenkins;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class AbstractDockerSlicerSpec
+        extends UnorderedStringSlicerSpec<AbstractProject<?, ?>> {
+
+    private static final Logger log = LoggerFactory.getLogger(AbstractDockerSlicerSpec.class);
+
+    static final String NOTHING = "(Nothing)";
```

Comment:
Generally we only use package private to make things visible to their tests. If that's what this is then I think it's fine